### DISCUSSION TITLE
MongoDB & Hibernate

### DISCUSSION THREAD

**Discussion by Tim203** [micronaut-projects/micronaut-data#2075] [Q&A] [answered]
Is it possible to write one repository & entity which then can be used for both MongoDB & Hibernate supported databases?
We don't know until runtime which database should be used and making two repo interfaces for each repo (one for Hibernate supported dbs and one for MongoDB) feels like it defeats the point of using an ORM like Hibernate.

I thought that I had it working with adding both the `Repository` and the `MongoRepository` annotation, but once I switched back from MongoDB to H2 I noticed that by adding MongoDB support I lost support for Hibernate (`org.hibernate.QueryException: unexpected char: '{' [{_id:{$eq:{$mn_qp:0}}}]`)

I also tried using Hibernate Reactive (because Vertx has a driver for JDBC and MongoDB) but couldn't get it to work.

**Comment by dstepanov** [comment, marked as answer]
You need to have separate repositories that are annotated MongoRepository and Repository, they can extend some common one to have shared code.
